core/server/worker/src/main/java/alluxio/worker/block/AsyncDeletionService.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.block;
+
+import alluxio.Sessions;
+import alluxio.exception.BlockDoesNotExistException;
+import alluxio.exception.InvalidWorkerStateException;
+import alluxio.util.ThreadFactoryUtils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.concurrent.ThreadSafe;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * Asychronous service to delete blocks.
+ */
+@ThreadSafe
+public class AsyncDeletionService {
+  private static final Logger LOG = LoggerFactory.getLogger(AsyncDeletionService.class);
+
+  private static final int DEFAULT_BLOCK_REMOVER_POOL_SIZE = 10;
+
+  private final BlockWorker mBlockWorker;
+  private final BlockingQueue<Long> mPendingRemovedBlocks;
+  private final Set<Long> mToAvoidDuplicateBlock;
```

Comment:
Is this just the set of blocks which (1) have been submitted and (2) are being removed? Maybe we can add a comment like that explaining the variable, and rename it to something like `mCurrentBlocks` or `mBlocks`.